profile/src/lib.rs

```diff
@@ -0,0 +1,192 @@
+use polars::prelude::*;
+use temp_name_lib::type_def::CLIGHT;
+use std::fs::File;
+use std::sync::Arc;
+pub struct config{
+    pub lambda_0:f64,
+    pub lambda_f:f64,
+    pub delta_lbd:f64,
+    pub v_max: f64,
+    pub n_phases:u32,
+}
+
+
+pub struct IntensityGrids{
+    pub grids:Vec<GridFeatures>,//<--effective temperature,log gravity;name_of_grid_file
+    pub lambda:Vec<f64>,//<-relevant wavelengths of grids
+
+    pub t_eff_useful_index: usize,
+    pub log_g_useful_index: usize,
+    pub lambda_useful_index: usize,
+}
+
+pub struct GridFeatures{
+    pub t_eff:f64,
+    pub log_g:f64,
+    pub file_name:String,
+}
+
+impl GridFeatures{
+    pub fn new(t_eff_value:f64,log_g_value:f64,file_name:&str)->Self{
+        GridFeatures { t_eff: t_eff_value,
+             log_g: log_g_value,
+             file_name: file_name.to_string() }
+    }
+
+}
+
+
+pub fn read_intensity_grid_file(path:&str) -> PolarsResult<LazyFrame> {
+    //let file = File::open(&path)?;
+
+    let schema =  Schema::from_iter(vec![
+		Field::new("wave_length".into(), DataType::Float64),
+		Field::new("a".into(), DataType::Float64),
+		Field::new("b".into(), DataType::Float64),
+		Field::new("c".into(), DataType::Float64),
+		Field::new("d".into(), DataType::Float64),
+		Field::new("ac".into(), DataType::Float64),
+		Field::new("bc".into(), DataType::Float64),
+		Field::new("cc".into(), DataType::Float64),
+		Field::new("dc".into(), DataType::Float64)
+    ]); 
+   
+    /*let parse_options = CsvParseOptions{
```

Comment:
Try to refrain from leaving commented out sections of code. For your successors this will cause problems as they don't know why it is commented out, or whether you forgot to uncomment it.